goalgorilla/open_social
# Feature/3113181 drupal composer update

## Change
Open Social is using the old composer scaffold plugin that will no longer be maintained for Drupal 9. The recommended method is to use the `core-composer-scaffold` package instead. The added benefit of that package is that it will copy files from installed composer packages instead of downloading each file one by one.

The cweagan/installers package is updated for compatibility with composer 2 and new Drupal related installer types that are supported.

## Issue tracker
https://www.drupal.org/project/social/issues/3113181

## How to test
Copy the following composer.json and run `composer install`. Besides this tests should pass. The composer.json below is a slightly modified version of the improved social_template.

```
{
    "name": "goalgorilla/social_template",
    "description": "Social project template for composer based Open Social projects.",
    "type": "project",
    "license": "GPL-2.0+",
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "php": "^7.2",
        "goalgorilla/open_social": "dev-feature/3113181-drupal-composer-update",
        "monolog/monolog": "^1.17"
    },
    "repositories": [
        {
            "type": "composer",
            "url": "https://packages.drupal.org/8"
        },
        {
            "type": "composer",
            "url": "https://asset-packagist.org"
        }
    ],
    "extra": {
        "drupal-scaffold": {
            "locations": {
                "web-root": "html/"
            }
        },
        "installer-types": [
            "bower-asset",
            "npm-asset"
        ],
        "installer-paths": {
            "html/core": [
                "drupal/core"
            ],
            "html/modules/contrib/{$name}": [
                "type:drupal-module"
            ],
            "html/profiles/contrib/social": [
                "goalgorilla/open_social"
            ],
            "html/profiles/contrib/{$name}": [
                "type:drupal-profile"
            ],
            "html/themes/contrib/{$name}": [
                "type:drupal-theme"
            ],
            "html/libraries/{$name}": [
                "type:drupal-library",
                "type:bower-asset",
                "type:npm-asset"
            ],
            "drush/contrib/{$name}": [
                "type:drupal-drush"
            ]
        },
        "enable-patching": true
    }
}
```

## Release notes
Open Social now uses the recommended `drupal/core-composer-scaffold` package to download scaffolding files. This replaces the outdated `drupal-composer/drupal-scaffold` package. Please check the change-record for details. 

## Change Record
Please review the following change record.

[ ] Agreed with the remarks about `drupal/drupal` vs `drupal/core` for framing of Open Social's role as package in a composer project.

https://www.drupal.org/node/3154837
